source/geometry_model/ellipsoidal_chunk.cc

```diff
@@ -130,23 +137,28 @@ namespace aspect
     Point<3>
     EllipsoidalChunk<dim>::EllipsoidalChunkGeometry::pull_back_ellipsoid(const Point<3> &x, const double semi_major_axis_a, const double eccentricity) const
     {
+      // The following converts x, y, z to phi, theta and negative depth
+      // Depth is measured perpendicular to the ellipsoid surface
+      // (i.e. along a vector which does not generally pass through the origin)
+      // Expressions can be found in Ellipsoidal and Cartesian Coordinates Conversion
+      // Subirana, Zornoza and Hernandez-Pajares, 2011:
+      // https://gssc.esa.int/navipedia/index.php/Ellipsoidal_and_Cartesian_Coordinates_Conversion
+
       AssertThrow (dim == 3,ExcMessage ("This can currently only be used in 3d."));
-      const double R    = semi_major_axis_a;
-      const double b      = std::sqrt(R * R * (1 - eccentricity * eccentricity));
-      const double ep     = std::sqrt((R * R - b * b) / (b * b));
-      const double p      = std::sqrt(x(0) * x(0) + x(1) * x(1));
-      const double th     = std::atan2(R * x(2), b * p);
-      const double phi    = std::atan2(x(1), x(0));
-      const double theta  = std::atan2(x(2) + ep * ep * b * std::pow(std::sin(th),3),
-                                       (p - (eccentricity * eccentricity * R  * std::pow(std::cos(th),3))));
-      const double R_bar = R / (std::sqrt(1 - eccentricity * eccentricity * std::sin(theta) * std::sin(theta)));
-      const double R_plus_d = p / std::cos(theta);
+      const double R    = semi_major_axis_a; // semi-major axis
```

Comment:
while you are at it, could you fix the indentation of the `=` at this line?

Reply:
Sure!